DeduplicateItems Fails on ion input

### Describe the issue

### Expected Behavior
The task should be able to deduplicate an ion file containing many messages based on a key field in those messages.

### Actual Behavior
For ion inputs, the task fails with the following error:
```
java.lang.IllegalArgumentException: [no message for java.lang.ClassCastException]
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4618)
	at com.fasterxml.jackson.databind.ObjectMapper.convertValue(ObjectMapper.java:4549)
	at io.kestra.core.tasks.storages.DeduplicateItems$PebbleFieldExtractor.extract(DeduplicateItems.java:199)
	at io.kestra.core.tasks.storages.DeduplicateItems$PebbleFieldExtractor.apply(DeduplicateItems.java:180)
	at io.kestra.core.tasks.storages.DeduplicateItems.run(DeduplicateItems.java:93)
	at io.kestra.core.tasks.storages.DeduplicateItems.run(DeduplicateItems.java:30)
	at io.kestra.core.runners.Worker$WorkerThread.run(Worker.java:769)
Caused by: com.fasterxml.jackson.databind.JsonMappingException: [no message for java.lang.ClassCastException]
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._wrapAsIOE(DefaultSerializerProvider.java:531)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider._serialize(DefaultSerializerProvider.java:504)
	at com.fasterxml.jackson.databind.ser.DefaultSerializerProvider.serializeValue(DefaultSerializerProvider.java:341)
	at com.fasterxml.jackson.databind.ObjectMapper._convert(ObjectMapper.java:4596)
```
	
This error does not appear if the file is serialized to JSON using `io.kestra.plugin.serdes.json.JsonWriter` prior to being passed to DeduplicateItems.

### Example Flow
```yml
# This works
  - id: json
    type: io.kestra.plugin.serdes.json.JsonWriter
    from: "{{ trigger.uri }}"

  - id: dedup
    type: io.kestra.core.tasks.storages.DeduplicateItems
    from: "{{ outputs.json.uri }}"
    expr: "{{ key }}"

#This does not
  - id: dedup
    type: io.kestra.core.tasks.storages.DeduplicateItems
    from: "{{ trigger.uri }}"
    expr: "{{ key }}"
```

### Environment

- Kestra Version: v0.16.2
- Operating System (OS/Docker/Kubernetes): Kubernetes / GKE v1.28.7-gke.1026000
- Java Version (if you don't run kestra in Docker): 17

--- a/core/src/main/java/io/kestra/core/tasks/storages/DeduplicateItems.java
+++ b/core/src/main/java/io/kestra/core/tasks/storages/DeduplicateItems.java
@@ -1,7 +1,6 @@
 package io.kestra.core.tasks.storages;
 
 import com.fasterxml.jackson.core.JsonProcessingException;
-import com.fasterxml.jackson.databind.JsonNode;
 import com.fasterxml.jackson.databind.ObjectMapper;
 import io.kestra.core.models.annotations.Example;
 import io.kestra.core.models.annotations.Plugin;
@@ -173,16 +172,6 @@ public class DeduplicateItems extends Task implements RunnableTask<DeduplicateIt
         private final RunContext runContext;
         private final String expression;
 
-        /** {@inheritDoc} */
-        @Override
-        public String apply(String data) throws Exception {
-            try {
-                return extract(MAPPER.readTree(data));
-            } catch (JsonProcessingException e) {
-                throw new RuntimeException(e);
-            }
-        }
-
         /**
          * Creates a new {@link PebbleFieldExtractor} instance.
          *
@@ -194,10 +183,20 @@ public class DeduplicateItems extends Task implements RunnableTask<DeduplicateIt
             this.expression = expression;
         }
 
-        public String extract(final JsonNode jsonNode) throws Exception {
-            @SuppressWarnings("unchecked")
-            Map<String, Object> map = MAPPER.convertValue(jsonNode, Map.class);
-            return runContext.render(expression, map);
+
+        /** {@inheritDoc} */
+        @Override
+        @SuppressWarnings("unchecked")
+        public String apply(String data) throws Exception {
+            try {
+                return extract(MAPPER.readValue(data, Map.class));
+            } catch (JsonProcessingException e) {
+                throw new RuntimeException(e);
+            }
+        }
+
+        public String extract(final Map<String, Object> item) throws Exception {
+            return runContext.render(expression, item);
         }
     }
 }

